### ISSUE TITLE
[a11y]: First cell in batch table header creating mild confusion for screen readers

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#10927]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

whichever is current on carbondesignsystem.com

### React version

7.54.2 https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

![image](https://user-images.githubusercontent.com/14298245/157062805-7f57aeec-ffd1-4f0e-9d32-dece45c64189.png)
^ Batch table with focus on Select All checkbox

The Select All checkbox in the first cell of the table header in a batch actions table is coded as a column header. As a result, its value is announced whenever the screen user moves to the first column in any row in table-reading mode (in JAWS) and when navigating (with Talkback and VoiceOver). Since the column header label is "Select All" the user hears (JAWS example)

> Select all rows
> column 1 row 2
> Select row check box not checked
> To check press Spacebar

This can be mildly confusing for the user, since it is quite easy to mistake the column header label as the label for the checkbox.

NOTE: It's not a problem when JAWS or is navigating by Tab between the checkboxes, because each of them is simply announced as "Select row checkbox not checked". The same is true of VO on a Mac when navigating via VO+arrow. 

The step to fix is technically simple:
- replace the TH of the Select All checkbox with a TD
- visually style it as if it were still a TH (to preserve the appearance of the table header row)
Existing
`<th scope="col" class="bx--table-column-checkbox"><div class="bx--checkbox--inline"><input class="bx--checkbox" id="data-table-1__select-all" name="select-all" type="checkbox"><label for="data-table-1__select-all" class="bx--checkbox-label" aria-label="Select all rows"></label></div></th>`
Modification
`<td class="bx--table-column-checkbox"><div class="bx--checkbox--inline"><input class="bx--checkbox" id="data-table-1__select-all" name="select-all" type="checkbox"><label for="data-table-1__select-all" class="bx--checkbox-label" aria-label="Select all rows"></label></div></td>`
![image](https://user-images.githubusercontent.com/14298245/157062286-4b0f08a2-47de-41bc-bfaf-a812d12c8779.png)
^ The style glitch appearing when the TH is removed and the TD is applied

On consideration, I rank this lower than originally thought, since in regular Tab navigation the table header does not get announced by JAWS, nor does it get announced by VoiceOver on Mac. Resolving this issue would improve the experience for JAWS users in Table Reading Mode and VO & TalkBack for mobile users.

### WCAG 2.1 Violation

It is a grey area. It is relevant to 1.3.1 Info & Relationships since the Select All checkbox is not a label for the column.

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Steps to reproduce

Navigate to https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

1. Activate JAWS or VoiceOver
2. Navigate to the Table (press T with Jaws)
3. With JAWS, turn on Table Reading (INS+Spacebar, T)
4. Use right arrow key to traverse the row to the Protocol column and then the down arrow key to go to the 3rd column
5. Use the left arrow key to return to the first column, listening for the table headers announced by JAWS.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
